### PR TITLE
Milestone 2: Query and update table

### PR DESCRIPTION
**Pull request by Rachelwebdev**
### In this Milestone, I:

- Added a new column `species` to the `animal` table
- Inserted more data into the previous one
- Began transactions where I updated and deleted some of the data in the table
- Wrote queries to specific questions based on the project requirements

**Below are the screenshot to the results of the queries**

QUERY 8
QUERY 9

QUERY 3
QUERY 4
QUERY 5
QUERY 6
QUERY 7

QUERY 15
QUERY 1
QUERY 2

Query 10
QUERY 11
Query 12
QUERY 13
QUERY 14

### CODE REVIEW

File: schema.sql [hunk 1]
```diff
@@ -4,5 +4,8 @@ CREATE TABLE animals(
     date_of_birth DATE NOT NULL,
     escape_attempts INT,
     neutered BOOLEAN NOT NULL,
-    weight_kg DECIMAL
+    weight_kg DECIMAL,    
 );
```
Comment (Gambit142) [inline schema.sql]: - [ ] I would advise you to remove the **_comma (,)_** after this line of code `weight_kg DECIMAL`. It is a bug that produces the syntax error message and consequently, the `animals` table can not be created. 😎 💻 

image

File: queries.sql [hunk 2]
```diff
@@ -5,4 +5,61 @@ SELECT date_of_birth FROM animals WHERE name = 'Agumon' OR name = 'Pikachu';
...
+DELETE FROM ANIMALS
+WHERE date_of_birth > '2022-01-01';
+
+UPDATE animals
+SET weight_kg = weight_kg * -1;
+
+UPDATE animals
+SET weight_kg = weight_kg * -1
+WHERE weight_kg < 0;
+
```
Comment (Gambit142) [inline queries.sql]: - [ ] When you want to start any transaction, kindly start with the `BEGIN` keyword. If you don't start with it, you will provoke a warning and Postgres will execute transactions in auto-commit mode. You can read this [phrase](https://www.postgresql.org/docs/current/sql-begin.html#:~:text=BEGIN%20initiates%20a,rollback%20is%20done) to understand better. Kindly start all transactions with the `BEGIN` keyword. 👍 😎

File: queries.sql [hunk 3]
```diff
@@ -5,4 +5,61 @@ SELECT date_of_birth FROM animals WHERE name = 'Agumon' OR name = 'Pikachu';
...
+DELETE FROM ANIMALS
+WHERE date_of_birth > '2022-01-01';
+
+UPDATE animals
+SET weight_kg = weight_kg * -1;
+
+UPDATE animals
+SET weight_kg = weight_kg * -1
+WHERE weight_kg < 0;
```
Comment (Gambit142) [inline queries.sql]: - [ ] Kindly make these one transaction based on the requirement. Also, please add a `SAVEPOINT` after the statement that deletes all animals born after **_Jan 1st, 2022_**.

### Project Requirement

image

#### Hint: Remember to start the transaction with a `BEGIN` statement. 👍 💻

**Review by Gambit142 — CHANGES_REQUESTED** [review]
Hi @Rachelwebdev 👋, 

Good job so far!

## Highlights
- Everything that creates or changes the table's structure is saved in the schema.sql file ✔️ 
- Everything that populates tables with data is saved in the data.sql file ✔️ 
- Everything that is a query that helps to analyze data is saved in the queries.sql file ✔️ 

There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Kindly check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

You can also consider:

N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

#### Please, remember to tag me in your question so I can receive the notification. You can tag me by typing @Gambit142

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

File: queries.sql (hunk 2 above)
Comment (Rachelwebdev) [reply queries.sql]: Thanks but I actually started all transactions with the BEGIN keyword, and it is reflected in my screenshots. I don't think my code would have worked without the use of the Keyword

**Review by TuyishimireEric — CHANGES_REQUESTED** [review]
# Status: APPROVED 🟢  
### Hi  @Rachelwebdev     👋 👋 

#### I'm @TuyishimireEric, Your code reviewer,
Good job on implementing the project requirements! 🥇 
done

Your project is complete!  There is nothing else to say other than... it's time to merge it 🦾  
Congratulations! 🎉

## Highlights
✔ correct SQL commands👍
✔ Made the suggested changes 👍

### Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/enus/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Rachelwebdev** [comment]
Thanks for the approval 👍

**Pull request merged**